rs/bitcoin/ckbtc/minter/src/updates/update_balance.rs

```diff
@@ -317,6 +328,9 @@ pub async fn update_balance<R: CanisterRuntime>(
     }
 
     schedule_now(TaskType::ProcessLogic, runtime);
+
+    let _ = &crate::metrics::UPDATE_CALL_LATENCY_WITH_NEW_UTXOS.with_borrow_mut(observe_latency);
```

Comment:
I think here we need a bit more granularity, because the latency is going to be greatly impacted by the number of processable UTXOs, since for each such UTXO:

1. If the UTXO is not ignored, we do 1 async call to the bitcoin checker
2. If the UTXO is not tainted, we do 1 async call the minter.

So I would say we need something like 
```rust
latency_per_num_utxos: BTreeMap<usize, LatencyHistogram>
```

This also has the advantage that the case of no new UTXOs is already covered (entry with key `0`) and does not need special handling. We could also refine the number of UTXOS between the ones that are ignored/tainted/minted but I'm not sure it's worth it at this point. WDYT?

Reply:
I increased the granularity of the metrics to keep track of the number of UTXOs in the `update_calls`, good idea! 

However, I'm wondering if there is actually a meaningful way to keep track of the distinction between ignored/tainted/minted UTXOs since there might be arbitrary combinations of each type within one call (e.g. 1 ignored, 2 tainted, 1 minted). I think maybe for now, as you said, we can keep it simple and potentially look into how we can track more detailed metrics later on.

Reply:
> I'm wondering if there is actually a meaningful way to keep track of the distinction between ignored/tainted/minted UTXOs since there might be arbitrary combinations of each type within one call (e.g. 1 ignored, 2 tainted, 1 minted).

Yes I think you would need some kind of counter struct
```rust
#[derive(Debug, Eq, PartialEq, PartialOrd, Ord)]
pub struct NumUtxos{
  num_ignored: usize,
  num_tainted: usize,
  num_minted: usize
}
```
it would need to implement `PartialOrd+Ord` to be able to be used a key in ` BTreeMap`.